### PR TITLE
CAS-490 clean up nexus "share" code

### PR DESCRIPTION
**Pull request by cjones1024**
- refactored "impl Nexus" to use Share trait implementation as appropriate
 - removed unused file: mayastor/src/bdev/nexus/nexus_iscsi.rs
 - removed unused file: mayastor/src/bdev/nexus/nexus_nvmf.rs
 - removed dead code from file: mayastor/src/target/nvmf.rs

Note that the ability to encrypt the share has been temporarily removed.
This needs to be revisited once the Share trait itself can accomodate such behaviour.

### CODE REVIEW

**Comment by cjones1024** [comment]
bors try

File: mayastor/src/bdev/nexus/nexus_share.rs [hunk 1]
```diff
@@ -52,49 +31,43 @@ impl Share for Nexus {
...
+                Err(Error::AlreadyShared {
                     name: self.name.clone(),
                 })
             }
         }
     }
 
     async fn unshare(&self) -> Result<Self::Output, Self::Error> {
-        self.bdev.unshare().await.map_err(|_e| Error::NotShared {
+        self.bdev.unshare().await.map_err(|_| Error::NotShared {
```
Comment (tiagolobocastro) [inline mayastor/src/bdev/nexus/nexus_share.rs:70]: I know it was already there, but that `Error::NotShared` does not seem correct?

File: mayastor/src/bdev/nexus/nexus_share.rs [hunk 2]
```diff
@@ -112,188 +85,87 @@ impl Share for Nexus {
...
-        };
-        self.share_handle = Some(name);
-        Ok(device_id)
+        }
     }
 
-    /// Undo share operation on nexus. To the chain of bdevs are all claimed
-    /// where the top-level dev is claimed by the subsystem that exports the
-    /// bdev. As such, we must first destroy the share and move our way down
-    /// from there.
```
Comment (tiagolobocastro) [inline mayastor/src/bdev/nexus/nexus_share.rs:242]: did you mean to replace this with another comment?

File: mayastor/src/core/bdev.rs [hunk 3]
```diff
@@ -51,6 +50,22 @@ pub struct BdevStats {
     pub bytes_written: u64,
 }
 
+impl From<iscsi::Error> for CoreError {
```
Comment (tiagolobocastro) [inline mayastor/src/core/bdev.rs:53]: should these be done using snafu .context instead?

File: mayastor/src/bdev/nexus/nexus_share.rs (hunk 2 above)
Comment (cjones1024) [reply mayastor/src/bdev/nexus/nexus_share.rs:242]: This pertains to the bdev encryption code that has been removed (and the comments with it) for now.

**Review by jonathan-teh — APPROVED** [review]

File: mayastor/src/bdev/nexus/nexus_share.rs (hunk 1 above)
Comment (cjones1024) [reply mayastor/src/bdev/nexus/nexus_share.rs:70]: Yeah, have created a new Snafu error type for this

File: mayastor/src/core/bdev.rs (hunk 3 above)
Comment (cjones1024) [reply mayastor/src/core/bdev.rs]: Decided to do things "properly" and use .context() everywhere for the sake of consistency.

File: mayastor-test/test_nexus.js [hunk 4]
```diff
@@ -157,7 +157,7 @@ function controlPlaneTest (thisProtocol) {
       (err, res) => {
         if (err) done(err);
         assert(res.device_uri);
-        if (thisProtocol === enums.NEXUS_NVMF) {
+        if (false /** FIXME - disable check as encryption is currently disabled */ && thisProtocol === enums.NEXUS_NVMF) {
```
[the remark this answers is not in the document]
Comment (cjones1024) [reply mayastor-test/test_nexus.js:160]: Agreed. Done

**Comment by cjones1024** [comment]
bors try

**Comment by cjones1024** [comment]
bors try

**Review by tiagolobocastro — APPROVED** [review]
:+1:

**Comment by cjones1024** [comment]
bors merge

**Pull request merged**